problem/problem_info.py

```diff
@@ -0,0 +1,29 @@
+import os.path
+from utils import config_info
+from problem.models import Problem, Testcase
+from django.db.models import Q
+
+SPECIAL_PATH = config_info.get_config('path', 'special_judge_path')
+PARTIAL_PATH = config_info.get_config('path', 'partial_judge_path')
+
+def get_testcase(problem):
+    problem.testcase = Testcase.objects.filter(problem=problem)
+    return problem
+
+def get_problem_list(user):
+    if user.is_anonymous():
+        return Problem.objects.filter(visible=True)
+    else:
+        if user.is_admin:
+            return Problem.objects.all()
+        else:
+            return Problem.objects.filter(Q(visible=True) | Q(owner=user))
+
+def has_special_judge_code(problem):
```

Comment:
We are going to add a column to record special / partial judge code language.
So you can use ```problem.judge_code_language``` to decide code file extension.